torch/distributed/pipelining/schedules.py

```diff
@@ -1951,6 +1953,73 @@ def backward_stage_index(step):
         )
 
 
+def _add_bubbles_to_actions(num_stages_global, pipeline_order, pp_group_size):
```

Comment:
What are the differences between this fn and https://github.com/pytorch/pytorch/blob/780b28f67ecd58a4e8c514562c260129cef0525d/torch/distributed/pipelining/schedules.py#L2100, could you comment it?

Reply:
Removed the duplicated function.

Reply:
@H-Huang IIUC this function is only needed so that the zbv schedule can be compatible with the preexisting PP runtime in PipelineScheduleMulti.  If we consolidate on the newer PipelineScheduleRuntime class, we don't have to add bubbles anymore and we can simplify our schedule IR generation.